server/service/src/vvm/vvm_status_log/insert/validate.rs

```diff
@@ -0,0 +1,49 @@
+use repository::{
+    vvm_status::{
+        vvm_status_log_row::{VVMStatusLogRow, VVMStatusLogRowRepository},
+        vvm_status_row::{VVMStatusRow, VVMStatusRowRepository},
+    },
+    RepositoryError, StockLineRow, StockLineRowRepository, StorageConnection,
+};
+
+use super::{InsertVVMStatusLogError, InsertVVMStatusLogInput};
+
+pub fn validate(
+    input: &InsertVVMStatusLogInput,
+    connection: &StorageConnection,
+) -> Result<(), InsertVVMStatusLogError> {
+    if check_vvm_status_log_exists(&input.id, connection)?.is_some() {
+        return Err(InsertVVMStatusLogError::VVMStatusLogAlreadyExists);
+    }
+
+    if check_vvm_status_exists(&input.status_id, connection)?.is_none() {
+        return Err(InsertVVMStatusLogError::VVMStatusDoesNotExist);
+    }
+
+    if check_stock_line_exists(&input.stock_line_id, connection)?.is_none() {
+        return Err(InsertVVMStatusLogError::StockLineDoesNotExist);
+    }
+
+    Ok(())
+}
+
+pub fn check_vvm_status_log_exists(
```

Comment:
This and below status check can just return a bool since the rows aren't being used anywhere 👀